web-services/annotations/src/main/java/datawave/webservice/annotation/Metadata.java

```diff
@@ -0,0 +1,103 @@
+package datawave.webservice.annotation;
+
+import java.util.List;
+import java.util.Objects;
+
+/**
+ * An immutable object that captures the internal Metadata associated with a record. It also provides a {@code equals} and {@code hashCode} implementation so
+ * that it can be used as a {@code Map} key. Similar to the {@link datawave.webservice.query.result.event.Metadata} class, but adds features specific to working
+ * with {@link datawave.annotation.protobuf.v1.Annotation} and {@link datawave.annotation.protobuf.v1.Segment}.
+ */
+public class Metadata {
+    /** the table where the record is stored */
+    private final String table;
+    /** the accumulo row / shard associated with a record */
+    private final String row;
+    /** the datatype of a record */
+    private final String dataType;
+    /** the internal identifier for a record */
+    private final String internalId;
+
+    /**
+     * Create a Metadata object using the field values from a {@link datawave.webservice.query.result.event.Metadata} object.
+     *
+     * @param eventMetadata
+     *            the object whose fields to copy.
+     */
+    public Metadata(datawave.webservice.query.result.event.Metadata eventMetadata) {
+        this(eventMetadata.getTable(), eventMetadata.getRow(), eventMetadata.getDataType(), eventMetadata.getInternalId());
+    }
+
+    /**
+     * Create a Metatata object based on the specified field values
+     *
+     * @param table
+     *            the table where the record is stored
+     * @param row
+     *            accumulo row / shard associated with a record
+     * @param dataType
+     *            the datatype of a record
+     * @param internalId
+     *            the internal identifier for a record
+     */
+    public Metadata(String table, String row, String dataType, String internalId) {
+        if (table == null || row == null || dataType == null || internalId == null) {
+            final String message = String.format("Null value provided when creating metadata: %s/%s/%s [%s]", row, dataType, internalId, table);
+            throw new NullPointerException(message);
+        }
+
+        this.row = row;
+        this.dataType = dataType;
+        this.internalId = internalId;
+        this.table = table;
+    }
+
+    /** Format metadata for error messages, since it doesn't have a reasonable {@code toString} method */
+    static String format(Metadata metadata) {
+        return String.format("%s/%s/%s [%s]", metadata.getRow(), metadata.getDataType(), metadata.getInternalId(), metadata.getTable());
```

Comment:
I think it'd be simpler to put this in `Metadata.toString()` and then whether you are printing a `Metadata` or a `List<Metadata>`, both should print nicely without needing a special call.

Reply:
Yes, I was being lazy when refactoring. Thanks for holding me accountable on this.